near-sdk/src/json_types/public_key.rs

```diff
@@ -217,13 +253,21 @@ mod tests {
     fn test_public_key_from_str() {
         let key = Base58PublicKey::try_from("ed25519:6E8sCci9badyRkXb3JoRpBj5p8C6Tw41ELDZoiihKEtp")
             .unwrap();
-        assert_eq!(key.0, binary_key());
+        assert_eq!(key, expected_key());
     }
 
     #[test]
     fn test_public_key_to_string() {
-        let key: Base58PublicKey = binary_key().try_into().unwrap();
+        let key: Base58PublicKey = expected_key();
         let actual: String = String::try_from(&key).unwrap();
         assert_eq!(actual, "ed25519:6E8sCci9badyRkXb3JoRpBj5p8C6Tw41ELDZoiihKEtp");
     }
+
+    #[test]
+    fn test_public_key_borsh() {
+        let key: Base58PublicKey = expected_key();
+        let encoded_key = key.clone().try_to_vec().unwrap();
+        let decoded_key = Base58PublicKey::try_from_slice(&encoded_key).unwrap();
+        assert_eq!(key, decoded_key);
+    }
```

Comment:
note: this test would not catch the format breaking change from this PR, would be nice to have serialized data (can just be in base58 format since those utils will be available in this module). Checking if it serializes symmetrically doesn't check much

Reply:
true, this was to just check if the manually implemented borsh (de)serialization worked as intended. But definitely having the serialized data to check for newer formats would be nice

Reply:
well it can work but not be in the same format as previously, which is a very breaking change and would break storage interactions when people migrate, which we can't have happen

Reply:
The newest version should have this where I created the original struct and check for the serialization between it and the current version to see if they're the same:
```Rust
    fn test_public_key_borsh_format_change() {
          // Original struct to reference Borsh serialization from
          #[derive(BorshSerialize, BorshDeserialize)]
          struct Base58PublicKeyRef(Vec<u8>);
```
What do you think?

Reply:
Ideally, the vector would just be the serialized bytes, which can be generated from `master` currently. This would test the correct format, it's just less strict.